base/shared/src/main/scala/scalaz/typeclass/LiskovInstances.scala

```diff
@@ -7,9 +7,6 @@ import Liskov.<~<
 
 trait LiskovInstances {
   /**We can witness equality by using it to convert between types */
-  implicit def witness[A, B](lt: A <~< B): A => B = {
-    type f[-X] = X => B
-    lt.subst[f](identity)
-  }
+  implicit def witness[A, B](lt: A <~< B): A => B = lt(_)
```

Comment:
Hmm. If you sign this `implicit def witness[A, B](a: A)(implicit lt: A <~< B): B = lt(a)` instead, does this compile?

```scala
def qqq[A, B](implicit lt: A <~< B): A => B = a => a
```

At the very least, an `implicit` keyword is missing here.

Reply:
Don't know, will try.

> At the very least, an implicit keyword is missing here.

I guess @aloiscochard meant this as an implicit conversion of `<~<` to function, not to provide an implicit functions.

Reply:
It compiles. That might make more sense. Note, however, that I didn't touch the signature, just the implementation.

Reply:
To a certain degree, those are the same thing now. I'm just not sure to what degree.

Reply:
@aloiscochard What do you think about changing `witness` to what @S11001001 proposed, i.e.

```scala
implicit def witness[A, B](a: A)(implicit lt: A <~< B): B = lt(a)
```

Passing `A <~< B` to a function that requires `A => B` could be achieved more simply by `Liskov` extending `Function1`.